### DISCUSSION TITLE
What happens to `components/ui` when there is a Chakra update?

### DISCUSSION THREAD

**Discussion by dan-scrumsdotcom** [chakra-ui/chakra-ui#9119] [General] [answered]
So far, I am loving the Shadcn inspiration and the new Chakra CLI. Thank you for the incredible efforts over the past years! I've been chatting to some of my colleagues about it, and we've often ended up on the following scenario and question:

---

Imagine, it's May 2026, you've built your production application in Chakra v3, and Chakra releases a v4. 

You've made some small changes to the components in the `ui` directory where they're auto added, to suite your project needs. 

Would I then be stuck on Chakra v3? Considering the code is all in my repository, I'm not fully understanding how Chakra would handle the updating of that code, if at all?

Many thanks for the insights 🙏

**Comment by segunadebayo** [comment, marked as answer]
Those files are yours. The CLI does not merge updates on upgrade.
